### ISSUE TITLE
extending CI testing

### ISSUE THREAD

**Issue by dvolgyes** [UCL/STIR#498]
Travis and AppVeyor do a good job, but they could be extended in several ways.
First of all, 'conda' based installation is not tested anywhere yet,
but it could be tested in Travis:
https://docs.conda.io/projects/conda/en/latest/user-guide/tasks/use-conda-with-travis-ci.html

Also, there are slight differences between various CI/CD systems, so testing the code
on multiple CI is a good practice. Circle CI is quite similar to Travis, 
and also supports conda installs, so STIR could extend its CI list with CircleCI, and both CircleCI and Travis could test the conda installs too.

**Comment by KrisThielemans** [comment]
Thanks for the suggestions. 

With SIRF, we have a limited effort on getting CI on a Jenkins server. The main advantage there is that would have access to MATLAB (as that is currently not CI tested). It should be feasible to extend this to STIR. However, I hear that support for PRs etc is a bit of a mess. @paskino could clarify.

I have no idea what would be necessary for CircleCI, or GitHub Actions. Contributions welcome. However, there needs to be a balance between debugging CI systems (often the case, downtimes etc), and actually testing the code. Also need to think about global warming :-)

Regarding conda installs, I'm not sure if I see the point:

- https://github.com/conda-forge/stir-feedstock already does testing for new conda builds. Admittedly, it doesn't run the cmake tests after the install (as it can't: it no longer has the build files), but it does run the `recon_test_pack`.
- the conda build gets up only updated very irregularly, and manually. There is no value in testing the conda-forge build after every commit in Travis.
- We could do a local conda-build of the recipe, but to me this would test how the integration with conda works, as opposed to STIR itself. For instance, I am struggling with https://github.com/conda-forge/stir-feedstock/pull/12 but not so sure if that's a STIR problem.

thoughts?

**Comment by dvolgyes** [comment]
I agree with the balance between testing the code and testing the CIs. 
That is a trade-off always.

conda: My point is that there are two parts at conda: building the code for conda, and testing if this build actually works (passes unit tests correctly).
I am not familiar with conda's testing infrastructure, but (rarely) i have seen conda packages which were deployable, but actually were wrong. But probably it is both complicated and overkill besides Travis. 

CircleCI: it is very much similar to Travis both in concept and in details.
(Same github login, etc.)
I don't promise it for short term, i am bit busy, but i will try to make a PR,
if nobody does it before me. 
Github actions: did not try it yet, but this seems to be the least likely to go bankrupt (financed by MS). I have more experience with CircleCI, but i might try this one too later. 

Jenkins remark: GitLab is also an alternative to Jenkins (it has self hosted verison, etc.) but it also offers free Travis-github like online services. Nothing against Jenkins, but it is also worth considering self hosted gitlab.
